apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -227,19 +279,60 @@ impl Plugin for Telemetry {
         &mut self,
         service: BoxService<RouterRequest, RouterResponse, BoxError>,
     ) -> BoxService<RouterRequest, RouterResponse, BoxError> {
+        let metrics = metrics::apollo::Metrics::default();
+        // TODO
+
+        self.apollo_metrics_sender.send(metrics);
         let metrics = BasicMetrics::new(&self.meter_provider);
         ServiceBuilder::new()
             .instrument(Self::router_service_span(
                 self.config.apollo.clone().unwrap_or_default(),
             ))
             .service(service)
+            .map_request(|req: RouterRequest| {
+                let client_name = req
+                    .originating_request
+                    .headers()
+                    .get("apollographql-client-name")
```

Comment:
Putting a todo here that the actual header name must come from the configuration since it can be overridden